### DISCUSSION TITLE
Question/remarque sur les fils pilotes

### DISCUSSION THREAD

**Discussion by drpixel972** [jmcollin78/versatile_thermostat#389] [Q&A]
Bonjour, 

Premièrement merci pour ton travail super intéressant.
Je poste en français mais si tu le souhaite, on peut basculer en anglais (ceci dit ... les fils pilotes sont un problème franco-français :-))
Idéalement dans l'utilisation du fil pilote on donne des ordres au radiateur.
J'ai chez moi du heatzy et des modules zibgee fil pilote. La logique vtherm pour eux visiblement pour utiliser le Vtherm serait de faire un switch on/off qui donne respectivement les ordres "confort"/"eco".
Je rate peut être un truc mais en suivant la logique le mode hors gel du vtherm ne sera cependant jamais atteignable car le mode éco est égal a confort - 3,5°C de mémoire. Du coup les 7° du mode hors gel ne seront jamais atteint car les radiateurs sont a minima mode éco.
Cela veut aussi dire que si une fenêtre reste ouverte trop longtemps, le radiateur même en mode éco va "chauffer" pour rien. 
Il faudrait que dans le cas d'un mode fil pilote, le vtherm en mode hors gel passe la consigne hors gel au fil pilote.
Le contournement serait d'utiliser le mode hors gel pendant le off (ou vraiment utiliser le off) au lieu d'utiliser le mode eco, mais je ne suis pas certain que ce soit pertinent en terme de consommation (ceci dit si vtherm fait bien le boulot ... la pièce ne sera jamais vraiment froide donc pourquoi pas ... néanmoins ça pose la question de la pertinence du fil pilote avec le vtherm vs un simple on/off ^^)
Est-il possible/envisageable que le vtherm descende les modes de consignes directement au hvac qu'il pilote en mode hvac ?
Si le vtherm est mis en mode "eco", il faudrait qu'il passe le hvac enfant en mode éco, idem pour les modes "confort" et "off".
Je reste à ta dispo pour discuter ou même faire des tests si nécessaire.

Sincères salutations,

**Comment by jmcollin78** [comment]
Hello @drpixel972 ,

Si tu regardes dans README et que tu penses avoir besoin d'un hors-gel, remplace simplement la consigne envoyée au radiateur par HG au lieu de ECO. Et ca devrait faire l'affaire. C'est là: https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#utilisation-dun-heatzy.

> Est-il possible/envisageable que le vtherm descende les modes de consignes directement au hvac qu'il pilote en mode hvac ?
Si le vtherm est mis en mode "eco", il faudrait qu'il passe le hvac enfant en mode éco, idem pour les modes "confort" et "off".

Ca existe, ça s'appelle le mode `over_climate`. En gros tu as une entité de type `climate` qui pilote ton radiateur et tu ajoutes un VTherm par dessus cette entité `climate`. Le VTherm va lui transmettre tous les ordres : chgt de preset, on, off et ça doit répondre à tes questions. Tu vas perdre en précision de régulation puisque c'est l'entité sous-jacente qui régule mais il y a des fonctions d'auto-regulation qui devraient aider.

**Reply by drpixel972** [reply]
Merci de ta réponse.
Pour le mode on/off la température ne descend pas trop en ce moment donc compliqué d'avoir un bon retour. Mais c'est ce que j'ai fait pour le moment.
Pour le mode over climate, je l'ai testé et le mode heat ou off fonctionne bien. Mais les presets ne fonctionnent pas. Le vtherm donne  l'ordre de température au climate sous-jacent ("ServiceCall climate.set_temperature ")
Sauf que les entités ne le supportent pas.

Pour ce genre d'entité, il faudrait conserver le fonctionnement du vtherm actuel mais les ordres de preset devraient faire : 
- confort --> service call hvac preset comfort
- eco --> service call hvac preset eco
- frost --> service call hvac preset away

Sur ce genre de device, la charge de régulation de la température doit rester sur le vtherm. Il devrait juste utiliser les modes hvac pour mettre en route ou pas le radiateur.
Qu'en penses-tu ? Au niveau option de configuration j'ai quelques pistes pour ne pas faire une usine à gaz. Reste à voir côté code la faisabilité technique sans faire de révolution.

Sincères salutations,

**Reply by jmcollin78** [reply]
Hello @drpixel972 ,

Je vois bien l'idée mais c'est totalement incompatible avec toutes les fonctions de régulation en mode `over_climate` qui régule par la température de consigne. Donc c'est tout un pan de fonctionnalités qui ne marcheront plus ou qui doivent être adaptées. Donc vraiment, je ne vais le faire (et en plus le mapping serait différent d'un device à un autre).

Pour le Heatzy, je te conseillerai de rester en mode `over_switch` (donc un pilotage par on/off) avec le on mappé sur Comfort et le Off sur 'Hors-gel'. Je pense que ca va faire tout ce que tu veux.

**Reply by drpixel972** [reply]
Merci du retour.
Je comprends bien la problématique s'il faut réadapter beaucoup de chose.
Pour le moment je vais rester en over_switch alors.
En tout cas merci pour ton sacré travail.

Sincères salutations,
